### PR TITLE
logs: changing config logs from JSON to YAML

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: Low
Testing: fixed up unit tests
Docs Changes: n/a
Release Notes: added

### CODE REVIEW

**Comment by moderation** [comment]
:frowning: the world needs less YAML

File: test/common/upstream/subset_lb_test.cc [hunk 1]
```diff
@@ -1686,10 +1686,10 @@ TEST_F(SubsetLoadBalancerTest, DescribeMetadata) {
   num_value.set_number_value(100);
 
   auto tester = SubsetLoadBalancerDescribeMetadataTester(lb_);
-  tester.test("version=\"abc\"", {{"version", str_value}});
+  tester.test("version=abc", {{"version", str_value}});
```
Comment (htuch) [inline test/common/upstream/subset_lb_test.cc:1689]: Do we have some better tests for this? One thing to note is that JSON is a proper subset of YAML, so any JSON tests for logging would pass today. Probably want some golden config fragment comparison.

**Comment by alyssawilk** [comment]
mm, I guess if some folks prefer json changing the default may be crummy?  I think adding a flag but having it not be the default would probably not achieve my goals.  Basically I thought we were trying to make yaml the gold standard for Envoy config, and it's annoying to take unit test config logs and have to manually convert them over before making them canonical config.

If we think having a yaml default and flag override is worthwhile I'll go add a command line flag, otherwise I'd be inclined to revert.  Thoughts?

**Comment by htuch** [comment]
If you're just after reformatting unit test configs to YAML, if we have the flag, you can run the test with `--test_arg` setting the flag.

**Comment by htuch** [comment]
Also, I've never tried it, but there's a bunch of utils like https://github.com/drbild/json2yaml.

**Comment by alyssawilk** [comment]
yeah, the server options are nowhere near the log lines, so I guess json wins :-(

**Pull request closed without merging**